<a href="https://colab.research.google.com/github/NontobekoNcube/NontobekoN/blob/main/SalesAnalysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

DATA ANALYSIS OF SALES DATA AT

In [12]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import statistics as stats

## 1. Load and Inspect data

In [4]:
from google.colab import files
uploaded = files.upload()
df = pd.read_csv('Sales.csv')


Saving Sales.csv to Sales.csv


### Understanding the data
This section explains the overall structure of the dataset

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7560 entries, 0 to 7559
Data columns (total 6 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Date    7560 non-null   object
 1   Time    7560 non-null   object
 2   State   7560 non-null   object
 3   Group   7560 non-null   object
 4   Unit    7560 non-null   int64 
 5   Sales   7560 non-null   int64 
dtypes: int64(2), object(4)
memory usage: 354.5+ KB


* The dataset has 7560 records(rows), and 6 colums(features).
* The columns include: Date, Time, State, Group, Unit, Sales.
* The data types in the dataset are integers and objects(strings).
* Nominal variables: Date,State, Group
* Ordinal variables: Time
* Discrete variables: Unit
* Continuous variables: Sales



In [7]:
df.head()


,Date,Time,State,Group,Unit,Sales
0,1-Oct-2020,Morning,WA,Kids,8,20000
1,1-Oct-2020,Morning,WA,Men,8,20000
2,1-Oct-2020,Morning,WA,Women,4,10000
3,1-Oct-2020,Morning,WA,Seniors,15,37500
4,1-Oct-2020,Afternoon,WA,Kids,3,7500


## 2. Location and Variability

In [18]:
print(f"Average sales amount: ${df['Sales'].mean():,.2f}")
print(f"Median units sold: {df['Unit'].median()}")
print(f"Most common state: {df['State'].mode()[0]}")  # [0] gets first result
print(f"Most popular demographic: {df['Group'].mode()[0]}")



Average sales amount: $45,013.56
Median units sold: 14.0
Most common state:  NSW
Most popular demographic:  Kids


## 3. Handle missing values

## 4. Covariance and Correlation

Univariate analysis

Bivariate analysis

Multivariate analysis

Detect Outliers

8. Feature Engineering

9. Summary of Insights